language/move-model/src/exp_rewriter.rs

```diff
@@ -67,6 +67,21 @@ impl<'env, 'rewriter> ExpRewriter<'env, 'rewriter> {
                 self.shadowed.pop_front();
                 res
             }
+            Quant(id, kind, ranges, body) => {
+                self.shadowed
+                    .push_front(ranges.iter().map(|(decl, _)| decl.name).collect());
+                let res = Quant(
+                    self.rewrite_attrs(*id),
+                    *kind,
+                    ranges
+                        .iter()
+                        .map(|(decl, range)| (decl.clone(), self.rewrite(range)))
```

Comment:
Actually, the "ranges" probably need to be computed before `self.shadowed.push_front(..)`

Reply:
(fixed)